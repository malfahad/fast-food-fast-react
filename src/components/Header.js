import React from 'react';

const Header = () => (
  <div className="header">
    <h1 className="heading main-title"> Fast Food Fast</h1>
    <h2 className="heading"> Order your food online </h2>
  </div>
);

export default Header;
