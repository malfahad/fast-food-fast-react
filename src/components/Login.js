import React from 'react';
import { connect } from 'react-redux';
import sampleAction from '../actions/sampleAction'

class Login extends React.Component {

    render(){
        return (<div>Login form</div>)
    }
}

const mapStateToProps = state=>({
    ...state
});

export default connect(mapStateToProps,{sampleAction})(Login)