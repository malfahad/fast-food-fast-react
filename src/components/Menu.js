import React from 'react';
import { connect } from 'react-redux';
import {
  AppHeader, OrderSummary, MenuForm, CustomerMenuButttons,
} from './Shared';
import fetchMenuAction from '../actions/menuAction';
import { addToOrder, removeFromOrder } from '../actions/orderActions';

export class Menu extends React.Component {
  componentWillMount() {
    this.props.fetchMenuAction();
  }

  onRemove = e => this.props.removeFromOrder(e.target.id);

  onAdd = e => this.props.addToOrder(e.target.id);

  render() {
    const { isAdmin } = this.props.user;
    const { menu, orderSummary, total } = this.props.menuState;
    const menuIds = Object.keys(menu);

    const menuItems = menuIds.map(key => (
      <div className="menu-item" key={key}>
        <img className="menu-item-img" src={menu[key].image_url} alt="menu item" />
        <h4 className="heading menu-item-title">
          {' '}
          {menu[key].title}
          {' '}
        </h4>
        <p>
          {' '}
          {menu[key].description}
          {' '}
        </p>
        <p className="item-price">
UGX
          {' '}

          {menu[key].amount}

        </p>
        <CustomerMenuButttons
          id={key}
          count={Object.prototype.hasOwnProperty.call(orderSummary, key)
            ? orderSummary[key].count : 0}
          onRemove={this.onRemove}
          onAdd={this.onAdd}
        />
      </div>
    ));

    const leftPane = isAdmin ? <MenuForm /> : (
      <OrderSummary
        orderSummary={Object.values(orderSummary)}
        total={total}
      />
    );

    return (
      <div>
        <AppHeader />
        <div className="container">
          {leftPane}
          <div className="column-main">
            <h3 className="heading">Restaurant Menu</h3>
            <p> This is the current fast food fast items menu.</p>
            <div id="menu">
              {menuItems}
            </div>
          </div>
        </div>
      </div>
    );
  }
}

const mapStateToProps = state => ({
  user: state.userReducer,
  menuState: state.menuReducer,
});

export default connect(mapStateToProps, {
  fetchMenuAction,
  addToOrder,
  removeFromOrder,
})(Menu);
