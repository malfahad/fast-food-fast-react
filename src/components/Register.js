import React from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import authAction from '../actions/authAction';

class Register extends React.Component {
  componentWillReceiveProps(nextProps) {
    if (nextProps.user.isLoggedIn) {
      localStorage.setItem('ff-token', nextProps.user.authorization);
      this.props.history.push('/');
    }
  }

    handleSubmit = (e) => {
      e.preventDefault();
      const fullName = e.target[0].value;
      const email = e.target[1].value;
      const password = e.target[2].value;
      const payload = {
        'full name': fullName,
        email,
        password,
      };
      this.props.authAction(false, payload);
    }

    render() {
      return (
        <div>
          <div className="header">
            <h1 className="heading main-title"> Fast Food Fast</h1>
            <h2 className="heading"> Order your food online </h2>
          </div>

          <div className="content">

            <form id="form-user-signup" onSubmit={this.handleSubmit} className="form-login" action="/">
              <h5 className="heading">Register a new account </h5>
              <p id="server-error" className="server-error">{this.props.user.authErrors}</p>
              <div className="form-group">
                <input type="text" className="form-control" placeholder="Full name" id="full_name" required />
              </div>
              <div className="form-group">
                <input type="email" className="form-control" placeholder="Email" id="email" required />
              </div>
              <div className="form-group">
                <input type="password" className="form-control" placeholder="Password" id="password" required />
              </div>
              <div className="form-group">
                <input type="password" className="form-control" placeholder="Repeat password" id="password2" required />
              </div>
              <div className="form-group">
                <input type="submit" className="form-control submit" value="Sign up" id="submit" />
              </div>
              <p>
                {' '}
Already registered?
                <Link to="/login">Sign in</Link>
              </p>

            </form>
          </div>
        </div>

      );
    }
}

const mapStateToProps = state => ({
  user: state.userReducer,
});

export default connect(mapStateToProps, { authAction })(Register);
