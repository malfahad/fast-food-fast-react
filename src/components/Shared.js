import React from 'react';

const Header = () => (
  <div className="header">
    <h1 className="heading main-title"> Fast Food Fast</h1>
    <h2 className="heading"> Order your food online </h2>
  </div>
);

export const AppHeader = () => (
  <div>

    <div className="heading-main">
      <h1 className="heading">Fast Food Fast</h1>
      <h3 className="heading">Order page</h3>
      <ul>
        <li className="btn btn-default logout"><a id="user-logout" href="login.html">Logout</a></li>
        <li className="btn btn-default logout"><a href="orders-history.html">Order history</a></li>
      </ul>
    </div>

    <div className="gap" />

  </div>
);

export const OrderSummary = props => (
  <div className="column-small persist">
    <h3 className="heading">Order Summary </h3>
    <ul id="order-summary">
      {props.orderSummary.map(item => (
        <li key={`${item.title}${item.count}`}>{`${item.count}x ${item.title}  - ${item.subTotal}`}</li>
      ))}
    </ul>
    <h4 id="order-total">{props.total > 0 ? `Total Ush ${props.total}` : ''}</h4>
    <button id="order-submit" type="button" className="menu-item-button" hidden={props.total === 0}>Submit Order</button>
  </div>
);

export const MenuForm = () => (
  <div className="column-small">
    <h3 className="heading">Add menu Item </h3>
    <form id="form-menu-item" method="post" action="/" encType="application/x-www-form-urlencoded">
      <p className="server-error" id="server-error" />
      <div className="form-group2 ">
        <input type="text" className="form-control2" placeholder="Menu Item Title" required id="menu-item-title" />
      </div>
      <div className="form-group2 ">
        <textarea type="text" className="form-control2" placeholder="Menu Item description" required id="menu-item-desc" />
      </div>
      <div className="form-group2 ">
        <input type="number" className="form-control2" placeholder="Amount" required id="menu-item-amount" />
      </div>
      <div className="form-group2 ">
        <small><label htmlFor="menu-item-img"> Menu Item Image: </label></small>
        <br />
        <input type="text" className="form-control2" placeholder="Menu Item Image url" id="menu-item-img" />
      </div>
      <div className="form-group2">
        <input type="submit" className="form-control submit" value="Create new" id="submit" />
      </div>
    </form>
  </div>
);

export const CustomerMenuButttons = props => (
  <div>
    <p className="item-counter">{props.count > 0 ? `${props.count} added` : ''}</p>
    <button id={props.id} type="button" className="menu-item-button" onClick={props.onAdd} hidden={props.count >= 10}>Add</button>
    <button id={props.id} type="button" className="menu-item-button" onClick={props.onRemove} hidden={props.count < 1}>Remove</button>
  </div>
);


export default Header;
