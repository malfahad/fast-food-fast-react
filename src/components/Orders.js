import React from 'react';
import { connect } from 'react-redux';
import { fetchOrders } from '../actions/orderActions';
import { AppHeader, OrderItem } from './Shared';

export class Orders extends React.Component {
  componentWillMount() {
    this.props.fetchOrders();
  }

    logout = (e) => {
      e.preventDefault();
      localStorage.removeItem('ff-token');
      const loginPath = (this.props.user.isAdmin) ? '/admin/login' : '/login';
      this.props.history.push(loginPath);
    }

    render() {
      const { orderHistory } = this.props;
      const orders = Object.values(orderHistory);
      orders.reverse();
      const orderItems = orders.length < 1 ? <p>No past orders</p> : (
        <div>
          {
          orders.map(order => (
            <OrderItem
              status={order.status}
              total={order.total}
              itemList={order.items}
            />
          ))
        }
        </div>
      );

      return (
        <div>
          <AppHeader
            logout={this.logout}
            title="order History"
          />
          <div className="container">
            <div className="column-wide">
              <h3 className="heading">My Order history</h3>
              <div id="orders-list">
                {orderItems}
              </div>
            </div>
          </div>
        </div>
      );
    }
}


const mapStateToProps = state => ({
  user: state.userReducer,
  menuState: state.menuReducer,
  orderHistory: state.ordersReducer.orders,
});

export default connect(mapStateToProps, {
  fetchOrders,
})(Orders);
