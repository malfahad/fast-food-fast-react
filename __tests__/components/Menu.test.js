import React from 'react';
import { shallow } from 'enzyme';
import { Menu } from '../../src/components/Menu';

describe('<Menu/>', () => {
  it('should render without crashing', () => {
    const props = {
      user: {
        isAdmin: false,
      },
      menuState: {
        menu: {},
        orderSummary: {},
        total: 0,
      },
      fetchMenuAction: jest.fn(),
    };
    const wrapper = shallow(<Menu {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render admin menu without crashing', () => {
    const props = {
      user: {
        isAdmin: true,
      },
      menuState: {
        menu: {},
        orderSummary: {},
        total: 0,
      },
      fetchMenuAction: jest.fn(),
    };
    const wrapper = shallow(<Menu {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should call addToOrder on add', () => {
    const props = {
      user: {
        isAdmin: false,
      },
      menuState: {
        menu: {
          1: {
            title: 'Fries',
            image_url: 'http://path.to/image',
            amount: 1000,
          },
        },
        orderSummary: {},
        total: 0,
      },
      fetchMenuAction: jest.fn(),
      addToOrder: jest.fn(),
      removeFromOrder: jest.fn(),
    };
    const wrapper = shallow(<Menu {...props} />);
    wrapper.instance().onAdd({ target: { id: 1 } });
    expect(wrapper.instance().props.addToOrder).toHaveBeenCalled();
  });

  it('should call removeFromOrder on remove', () => {
    const props = {
      user: {
        isAdmin: false,
      },
      menuState: {
        menu: {
          1: {
            title: 'Fries',
            image_url: 'http://path.to/image',
            amount: 1000,
          },
        },
        orderSummary: {
          1: {
            count: 1,
          },
        },
        total: 0,
      },
      fetchMenuAction: jest.fn(),
      addToOrder: jest.fn(),
      removeFromOrder: jest.fn(),
    };
    const wrapper = shallow(<Menu {...props} />);
    wrapper.instance().onRemove({ target: { id: 1 } });
    expect(wrapper.instance().props.removeFromOrder).toHaveBeenCalled();
  });
});
