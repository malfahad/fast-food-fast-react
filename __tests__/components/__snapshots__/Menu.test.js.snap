// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Menu/> should render admin menu without crashing 1`] = `ShallowWrapper {}`;

exports[`<Menu/> should render without crashing 1`] = `ShallowWrapper {}`;
